File: Kwetter/src/main/java/service/PostService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package service;

import dao.PostDao;
import domain.Post;
import java.util.List;
import javax.ejb.Stateless;
import javax.inject.Inject;
import service.exceptions.NonExistingPostException;
import service.exceptions.NonExistingUserException;

/**
 *
 * @author Tom
 * @email
 * @version 0.0.1
 */
@Stateless
public class PostService {

    @Inject
    PostDao postDao;

    @Inject
    ServiceExceptionHandler exh;

    public List<Post> getAllPosts() {
        return postDao.getAllPosts();
    }

    public List<Post> getPostsByPoster(long userId) throws NonExistingUserException {
        exh.NonExistingUserCheck(userId);
        return postDao.getPostsByPoster(userId);
    }

    //TODO Add max-min limit
    public List<Post> getRecentPostsByPoster(long userId, int limit) throws NonExistingUserException {
        exh.NonExistingUserCheck(userId);
        return postDao.getRecentPostsByPoster(userId, limit);
    }

    public long getPostCountByPoster(long userId) throws NonExistingUserException {
        exh.NonExistingUserCheck(userId);
        return postDao.getPostCountByPoster(userId);
    }

    public List<Post> searchPosts(String input) {
        return postDao.searchPosts(input);
    }

    public List<Post> getTimeline(long userId, int limit) throws NonExistingUserException {
        exh.NonExistingUserCheck(userId);
        return postDao.getTimeline(userId, limit);
    }

    public Post createNewPost(long userId, String content) throws NonExistingUserException {
        exh.NonExistingUserCheck(userId);
        return postDao.createNewPost(userId, content);
    }

    public void deletePost(long postId) throws NonExistingPostException {
        exh.NonExistingPostCheck(postId);
        postDao.deletePost(postId);
    }

    void setExceptionHandler(ServiceExceptionHandler exh) {
        this.exh = exh;
    }

    void setPostDao(PostDao postDao) {
        this.postDao = postDao;
    }
}

File: Kwetter-App/src/app/services/post/post.service.ts
import {Injectable} from '@angular/core';
import {ApiService} from '../api/api.service';
import {Post} from '../../models/post';

@Injectable()
export class PostService {

    postUrl: string = "posts";

    constructor(private apiService: ApiService) {}

    getRecentPostsByUser(userId: number, limit: number): any {
        let url: string = this.postUrl + "/getRecentPostsByPoster/" + userId + "/" + limit;
        return this.apiService.get<any>(url);
    }

    createNewPost(userId: number, content: string): any {
        let url: string = this.postUrl + "/createNewPost/"
        return this.apiService.post<any>(url, {"posterId": userId, "content": content}, true);
    }

    getTimeline(userId: number, limit: number): any {
        let url: string = this.postUrl + "/getTimeline/" + userId + "/" + limit;
        return this.apiService.get<any>(url);
    }

    searchPosts(searchCriteria: string): any {
        let url: string = this.postUrl + "/searchPosts/" + searchCriteria;
        return this.apiService.get<any>(url);
    }
    
    
}
